lib/logstash/codecs/cef.rb

```diff
@@ -103,14 +103,31 @@ def decode(data)
         message=message + ' ' unless message.end_with?('\=')
       end
 
-      # Now parse the key value pairs into it
-      extensions = {}
-      message = message.split(/ ([\w\.]+)=/)
-      key, value = message.shift.split('=', 2)
-      extensions[key] = value.gsub(/\\=/, '=').gsub(/\\\\/, '\\')
-      Hash[*message].each{ |k, v| extensions[k] = v }
-      # And save the new has as the extensions
-      event.set('cef_ext', extensions)
+      # Introducing custom delimiters instead of space to avoid convolution when parsing fields with ' ' spaces.
+      # Example : cs1Label=New Malware cs1=Conficker ==> cs1Label=New Malware|^^^cs1=Conficker
+      message=message.gsub((/\s+(\w+=)/)||(/(\w+=)/),'|^^^\1')
+
+      # Truncating the additional fields, if needed it can be mapped from ArcSight's connector; otherwise also protects the dot (.) notations in implmentations for Elasticsearch v 2.4 and below
+      # Example ad.field[0]=XXXX ad.name[2]=abc
+      message=message.gsub((/ (ad.\w+.*\]\=[^|^^^]+)/),'')
+
+      # Splits the fields by the custom delimiter introduced above
+      message=message.split('|^^^')
+
+      # Introducing a custom delimiter to avoid convolution when parsing fields with '=' equals
+      # Example : cs4=https://<sampledomain>/event_stream/events_for_bot?inc_id\\=50; requestUrl=http://db:3301/q='select * from a where b=c'
+      for i in 0..message.length-1
+        message[i]=message[i].sub(/\=/, "***")
```

Comment:
Like how the custom field name delimiter was introduced, this is a custom separator where given probable values in the KV might contain '='. For instance:

> cs1=Internet Explorer 8.x cs2=http://abc.domain?q=some_query

On parsing cs2, tokenization might cause cs2 to only carry the value of http://abc.domain? and q=some_query to be broken up to another field name. 

The *** is just a mechanism to safeguard to properly parse absolute values as a whole beyond the delimiter.